extensions/amp-mustache/0.1/amp-mustache.js

```diff
@@ -33,14 +36,37 @@ export class AmpMustache extends AMP.BaseTemplate {
 
   /** @override */
   compileCallback() {
+    /** @private @const {!JsonObject} */
+    this.nestedTemplates_ = dict();
+    const shadowTemplate = this.element.cloneNode(true);
+    let index = 0;
+    iterateCursor(templateContent(shadowTemplate).querySelectorAll('template'),
+        nestedTemplate => {
+          const nestedTemplateKey = `__AMP_NESTED_TEMPLATE_${index}`;
+          this.nestedTemplates_[nestedTemplateKey] =
+              nestedTemplate./*OK*/outerHTML;
+
+          const nestedTemplateAsVariable = this.element.ownerDocument
+              .createTextNode(`{{{${nestedTemplateKey}}}}`);
```

Comment:
Wooo boy, unsafe template insertion. Has the been security reviewed?

Reply:
It looks ok to me - all templates are sanitized. This is a strictly text node so no XSS vector.

Reply:
But it will be inserted later by mustache using unsafe `{{{variable}}}` syntax.

Reply:
Hasn't been reviewed, but if you look later in `render(...)` I drop any key that the users pass that start with `__AMP_NESTED_TEMPLATE_` (which they shouldn't pass to begin with) - should we still get this reviewed?

Reply:
Removed the unnecessary filter in the render method and added a test to ensure that the nested templates get sanitized when being rendered

Reply:
What's the actual flow? Does the overall template, including all nested resolutions evaluated and then the final sanitizer pass is applied? Or are they all done separately?

Alternatively, we can just mark the production templates and clone the original content into them if that feels like would be safer?